content/concepts/protocols.md

```diff
@@ -0,0 +1,268 @@
+This can be useful if you support a range of protocol versions, since you can propose the most recent version and fallback to older versions
+if the remote hasn't adopted the latest version yet.
+
+## Core libp2p protocols
+
+In addition to the protocols that you write when developing a libp2p application, libp2p itself defines several foundational protocols that are used for core features.
+
+### Common patterns
+
+The protocols described below all use [protocol buffers](https://developers.google.com/protocol-buffers/) (aka protobuf) to define message schemas.
+
+Messages are exchanged over the wire using a very simple convention called [msgio](https://github.com/jbenet/go-msgio), which prefixes binary
+message payloads with an integer that represents the length of the payload in bytes. The length is encoded as a [protobuf varint](https://developers.google.com/protocol-buffers/docs/encoding#varints)  (variable-length integer).
+
+
+<!-- TODO(yusef): do we expose any kind of msgio + protobuf convenience functions for protocol construction? If so, link here. also, is there a spec for msgio in the abstract? -->
+
+### Ping
+
+| **Protocol id**    | spec |               |               | implementations   |
+|--------------------|------|---------------|---------------|-------------------|
+| `/ipfs/ping/1.0.0` | N/A  | [go][ping_go] | [js][ping_js] | [rust][ping_rust] |
+
+
+[ping_go]: https://github.com/libp2p/go-libp2p/tree/master/p2p/protocol/ping
+[ping_js]: https://github.com/libp2p/js-libp2p-ping
+[ping_rust]: https://github.com/libp2p/rust-libp2p/blob/master/protocols/ping/src/lib.rs
+
+
+The ping protocol is a simple liveness check that peers can use to quickly see if another peer is online.
+
+After the initial protocol negotiation, the dialing peer sends 32 bytes of random binary data. The listening
+peer echoes the data back and closes the stream, and the dialing peer will verify the response and measure 
+the latency between request and response.
+
+### Identify
+
+| **Protocol id**  | spec                           |                   |                   | implementations       |
+|------------------|--------------------------------|-------------------|-------------------|-----------------------|
+| `/ipfs/id/1.0.0` | [identify spec][spec_identify] | [go][identify_go] | [js][identify_js] | [rust][identify_rust] |
+
+<!-- TODO(yusef): update spec link on PR merge -->
+[spec_identify]: https://github.com/libp2p/specs/pull/97/files
+[identify_go]: https://github.com/libp2p/go-libp2p/tree/master/p2p/protocol/identify
+[identify_js]: https://github.com/libp2p/js-libp2p-identify
+[identify_rust]: https://github.com/libp2p/rust-libp2p/tree/master/protocols/identify/src
+
+The `identify` protocol allows peers to exchange information about each other, and serves a crucial role
+in [NAT traversal](/concepts/nat/).
```

Comment:
that's probably overstating `identify`'s role in nat traversal, which is minimal.